client/components/TitleEditor/TitleEditor.tsx

```diff
@@ -0,0 +1,185 @@
+import classNames from 'classnames';
+import { useBeforeUnload } from 'react-use';
+import React, { ClipboardEvent, useCallback, useEffect, useRef, useState } from 'react';
+
+require('./titleEditor.scss');
+
+type Props = {
+	isReadOnly?: boolean;
+	initialValue?: string;
+	onChange?: (value: string) => void;
+	className?: string;
+	placeholder?: string;
+};
+
+const SUPPORTED_DECORATIONS = new Set(['i', 'em', 'b', 'strong']);
+
+function isChildOf(descendant: Node, ancestor: Node) {
+	let node: Node | null = descendant;
+	while (node !== null) {
+		if (node === ancestor) return true;
+		node = node.parentNode;
+	}
+	return false;
+}
+
+function stripAttrs(element: Element) {
+	while (element.attributes.length > 0) {
+		element.removeAttribute(element.attributes[0].name);
+	}
+}
+
+function sanitizeElement(element: Element) {
+	if (!SUPPORTED_DECORATIONS.has(element.tagName.toLowerCase())) {
+		const parent = element.parentNode!;
+		if (element.textContent !== null && element.textContent.length > 0) {
+			const textNode = document.createTextNode(element.textContent ?? '');
+			parent.replaceChild(textNode, element);
+		} else {
+			parent.removeChild(element);
+		}
+		return;
+	}
+	stripAttrs(element);
+	if (element.nodeType === Node.ELEMENT_NODE) {
+		for (let i = element.children.length - 1; i >= 0; i--) {
+			sanitizeElement(element.children[i]);
+		}
+	} else if (element.nodeType === Node.TEXT_NODE) {
+		element.textContent = element.textContent?.replace(/[\r\n]+/gm, '') ?? null;
+	}
+}
+
+function sanitizeDocumentFragment(doc: DocumentFragment) {
+	for (let i = doc.children.length - 1; i >= 0; i--) {
+		sanitizeElement(doc.children[i]);
+	}
+}
+
+function parseDom(html: string) {
+	const template = document.createElement('template');
+	template.innerHTML = html;
+	return template;
+}
+
+const commonProps = {
+	role: 'textbox',
+	'aria-label': 'Edit Pub title',
+	tabIndex: 0,
+};
+
+export default function TitleEditor(props: Props) {
+	const { initialValue = '', onChange, isReadOnly = false, ...restProps } = props;
+	const node = useRef<HTMLDivElement>(null);
```

Comment:
It's blowing my mind that TypeScript doesn't complain about this...I would have thought you'd have to put `null | HTMLDivElement` in the brackets in order to set the initial value to `null`.